src/MIDebugEngine/AD7.Impl/AD7StackFrame.cs

```diff
@@ -399,6 +399,15 @@ int IDebugStackFrame2.EnumProperties(enum_DEBUGPROP_INFO_FLAGS dwFields, uint nR
                     guidFilter == AD7Guids.guidFilterLocals ||
                     guidFilter == AD7Guids.guidFilterLocalsPlusArgs)
                 {
+                    /*
+                    Question: do we want to filter on guidFilter?
+
+                    if (guidFilter == AD7Guids.guidFilterLocalsPlusArgs)
+                    {
+                        _hasGottenLocalsAndParams = false;
+                    }
+                    */
+                    _hasGottenLocalsAndParams = false;
```

Comment:
>_hasGottenLocalsAndParams = false; [](http://example.com/codeflow?start=20&length=34)

This change makes `_hasGottenLocalsAndParams`/`_parameters`/`_locals` useless as a cache -- we always recalcuale them. So, assuming this is the right way to go (which I think is true), we should get rid of those member variables.

I wouldn't expect that the UI would repeatedly call EnumProperties. So I think ditching the cache is probably the way to go. Is that what you saw?

Reply:
As long as you aren't seeing the locals repeatedly queried in cases where we should be able to cache things -- yes, I think it makes sense to just delete the cache. The other option would be to change your fix to invalidate the cache when particular things happened, like maybe when we evaluate from the immediate window that should invalidate. But I think I am of the opinion that we don't really know any better than our caller when things have changed, so we shouldn't be caching.

Reply:
I can't find a viable way to remove `_parameters` and `_locals`. Do you have any suggestions for how to do so? If I remove both arrays and their associated functions completely, the Locals window will be completely blank.

Reply:
You want to change the code to:
1. Delete this `if` block that you are changing
2. Rename `EnsureLocalsAndParameters()` to `GetLocalsAndParameters(out List<VariableInformation> parameters, out List<VariableInformation> locals)`
3. Change all the `CreateXYZProperties` functions to call `GetLocalsAndParameters` instead of using `_parameters` and `_locals`
4. Remove all the member variables.

Reply:
I have been trying to make the `_parameters` and `_locals` changes without the Locals window going blank to no avail:
![image](https://user-images.githubusercontent.com/60245970/161300912-75762ad7-68c1-4cc6-9dbf-d18e61fd329c.png)

Will investigate in the meantime.

Reply:
Update: I fixed the blank Locals window issue. However, I had to remove this check in `GetLocalsAndParameters` to do so: `if (_textPosition != null && radix != _radix).
`